src/makeInfinityAdder.js

```diff
@@ -4,7 +4,21 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let amount = 0;
+
+  return function adder(number) {
+    if (number === undefined) {
+      const result = amount;
+
+      amount = 0;
+
+      return result;
+    }
+
+    amount += number;
+
+    return adder;
```

Comment:
The `adder` function correctly implements the required functionality of summing the numbers and resetting the amount when called without parameters. However, you might consider renaming the parameter `number` to something more descriptive, like `valueToAdd`, to enhance code readability.